sse/event_source.go

```diff
@@ -38,23 +38,47 @@ type EventSource struct {
 
 	currentReadCloser *ReadCloser
 	lastEventID       string
-	retryInterval     time.Duration
 	lock              sync.Mutex
 
-	closeOnce *sync.Once
-	closed    chan struct{}
+	closeOnce   *sync.Once
+	closed      chan struct{}
+	retryParams RetryParams
```

Comment:
As much as I appreciate grouping together related constructs (interval + limit), I don't like the reuse of this struct between input config + state on the SSE object, as before it was purely to track the interval configured by the event stream itself.

Perhaps just flatten it again and have the constructor initialize `retryInterval` to the configured one if it's nonzero, and otherwise `defaultRetryInterval`? So that'd be keeping the struct as the input param, but keeping the private field as well, and possibly just putting `maxRetries` next to it.